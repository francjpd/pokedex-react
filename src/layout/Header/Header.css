.pokedex-header {
  display: grid;
  & .pokedex-header__content {
    background-color: var(--secondary-color);
    & img {
      width: 130px;
      padding: 10px;
      margin: auto;
      display: flex;
    }

    & .pokedex-header__menu {
      background-color: var(--secondary-color);
      & button {
        background-color: transparent;
        color: white;
        font-family: unset;
        border-radius: 0;
      }
    }
  }
}
