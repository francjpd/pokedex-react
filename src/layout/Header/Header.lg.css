@media (--lg) {
  .pokedex-header {
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
    & .pokedex-header__content {
      grid-column: 3/11;
    }
  }
}
