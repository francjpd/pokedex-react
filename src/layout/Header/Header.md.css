@media (--md) {
  .pokedex-header {
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
    & .pokedex-header__content {
      grid-column: 1/9;
    }
  }
}
