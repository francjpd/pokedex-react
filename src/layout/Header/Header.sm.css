@media (--sm) {
  .pokedex-header {
    grid-template-columns: 1fr 1fr 1fr 1fr;
    & .pokedex-header__content {
      grid-column: 1/5;
    }
  }
}
