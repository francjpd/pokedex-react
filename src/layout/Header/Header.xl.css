@media (--xl) {
  .pokedex-header {
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
    & .pokedex-header__content {
      grid-column: 4/10;
    }
  }
}
