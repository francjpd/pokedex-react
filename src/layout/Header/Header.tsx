
import headerSvg from '../../assets/header/header.svg';
import './Header.css';
import './Header.sm.css';
import './Header.md.css';
import './Header.lg.css';
import './Header.xl.css';
import { useLocation, useNavigate } from 'react-router-dom';

export default function Header() {

    const navigate = useNavigate();
    const location = useLocation();
    return (
        <div className='pokedex-header'>
            <div className='pokedex-header__content'>
                <img src={headerSvg} alt='pokemon logo' />
                <div className='pokedex-header__menu'>
                    {location.pathname !== '/' &&
                        <button onClick={() => navigate('/')}> List</button>
                    }
                    <button onClick={() => navigate('/add')}> Add Pokemon</button>
                </div>
            </div>
        </div>
    )
}