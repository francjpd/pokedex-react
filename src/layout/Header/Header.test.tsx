import { render, screen } from '@testing-library/react';
import Header from './Header';
jest.mock('react-router-dom')

describe('Headers Layout test', () => {
    let component: any;
    beforeEach(() => {
        component = render(<Header />);
    })
    it('should render the Header layout correctly', () => {
        expect(component).toMatchSnapshot();

        expect(screen.getByText(/Add Pokemon/i)).toBeInTheDocument()
        expect(screen.queryByText(/List/i)).toBeNull()
    });

})
