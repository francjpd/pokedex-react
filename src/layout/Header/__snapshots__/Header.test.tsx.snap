// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headers Layout test should render the Header layout correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pokedex-header"
      >
        <div
          class="pokedex-header__content"
        >
          <img
            alt="pokemon logo"
          />
          <div
            class="pokedex-header__menu"
          >
            <button>
               Add Pokemon
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pokedex-header"
    >
      <div
        class="pokedex-header__content"
      >
        <img
          alt="pokemon logo"
        />
        <div
          class="pokedex-header__menu"
        >
          <button>
             Add Pokemon
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
